add ripple, disabled.  use renderer.

jefrog1844 opened the pull request:
Please review

amorey commented on projects/mui-angular/button/button.component.ts:
```diff
@@ -1,36 +1,82 @@
-    <button class="mui-btn" #button>
+    <button class="mui-btn" #button [disabled]="disabled">
       <ng-content></ng-content>
+      <span class="mui-btn__ripple-container" #container><span class="mui-ripple" #ripple></span></span>
```
Is the `#container` reference being used?

jefrog1844 replied:
No, will remove.

amorey commented on projects/mui-angular/button/button.component.ts:
```diff
@@ -1,36 +1,82 @@
+
+  @HostListener('mousedown', ['$event', '$event.currentTarget'])
+  mouseDown(event: MouseEvent) {
+    if(this.disabled) return;
```
```suggestion
    if (this.disabled) return;
```

jefrog1844 replied:
Done

amorey commented on projects/mui-angular/button/button.component.ts:
```diff
@@ -1,36 +1,82 @@
+    var offsetWidth, offsetHeight, offsetTop, offsetLeft, radius, diameter, width, height, left, top;
+
+    // get (x, y) position of click
+    offsetWidth = this.button.nativeElement.offsetWidth;
```
You can save some keystrokes by saving references to `this.button.nativeElement` and `this.ripple.nativeElement`

jefrog1844 replied:
added private variables and set in ngAfterViewInit

amorey commented on projects/mui-angular/button/button.component.ts:
```diff
@@ -1,36 +1,82 @@
       let attrVal = this[attrName];
       if (attrVal) {
-        this.button.nativeElement.classList.add('mui-btn--' + attrVal);
+        this.renderer.addClass(this.button.nativeElement, 'mui-btn--' + attrVal);
```
👍 Woot! The renderer makes class manipulation much easier.

jefrog1844 replied:
even nicer with cached elements!

amorey commented:
Looks great! I added some suggestions in-line. Two other comments:

- If possible it would be nice to add some tests as well. I've found that even simple tests can help you to improve your code and make it more maintainable in the long-run.
- Are you familiar with [Airbnb's JavaScript styleguide](https://github.com/airbnb/javascript)? I use it for my javascript code and have found it to be very useful. It has some tips that would be helpful here too.

amorey commented on projects/mui-angular/button/button.component.ts:
```diff
@@ -1,36 +1,82 @@
   @ViewChild('button') button: ElementRef;
+  @ViewChild('ripple') ripple: ElementRef;
 
   @Input() variant: string = '';
   @Input() color: string = '';
   @Input() size: string = '';
```
Should we make these inputs `optional?` as well?

jefrog1844 replied:
Yes.

jefrog1844 commented:
Got your feedback and I will submit an update based on your feedback.

1. Is there an editor you use that has the airbnb styleguide incorporated?
2. Are you fine with the [disabled]="true"?  To use "disabled" instead of
"[disabled]=true], I would have to use a set method on the input with
another private const to bind the disabled tag to?

Thanks

Jeff

On Wed, Sep 9, 2020 at 2:06 AM Andres Morey <notifications@github.com>
wrote:

> Looks great! I added some suggestions in-line. Two other comments:
>
>    - If possible it would be nice to add some tests as well. I've found
>    that even simple tests can help you to improve your code and make it more
>    maintainable in the long-run.
>    - Are you familiar with Airbnb's JavaScript styleguide
>    <https://github.com/airbnb/javascript>? I use it for my javascript
>    code and have found it to be very useful. It has some tips that would be
>    helpful here too.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/muicss/mui-angular/pull/2#issuecomment-689353881>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM6FHO4DRNBME4GQ7ELZJJTSE4SO7ANCNFSM4RA46NZQ>
> .
>

amorey commented:
For help with the styleguide, it looks like you can use eslint to do Airbnb styleguide linting from the command line:
https://www.npmjs.com/package/eslint-config-airbnb

You can also integrate the linter with some popular IDE's:
- Atom (https://www.acuriousanimal.com/blog/2016/08/14/configuring-atom-with-eslint)
- SublimeText (https://www.cuga-moylan.com/blog/setting-up-eslint-with-airbnb-styleguide-in-sublime-text-3/)

For the pull request, if you push your changes to your github repo then it should update this PR automatically.

jefrog1844 commented:
I have the changes done and have written the test class, but I’m not able to get the test classes to run except those in the src directory that are the generic component and service you created.  Any ideas?

Thanks

Jeff

> On Sep 9, 2020, at 4:05 PM, Andres Morey <notifications@github.com> wrote:
> 
> 
> For help with the styleguide, it looks like you can use eslint to do Airbnb styleguide linting from the command line:
> https://www.npmjs.com/package/eslint-config-airbnb <https://www.npmjs.com/package/eslint-config-airbnb>
> You can also integrate the linter with some popular IDE's:
> 
> Atom (https://www.acuriousanimal.com/blog/2016/08/14/configuring-atom-with-eslint <https://www.acuriousanimal.com/blog/2016/08/14/configuring-atom-with-eslint>)
> SublimeText (https://www.cuga-moylan.com/blog/setting-up-eslint-with-airbnb-styleguide-in-sublime-text-3/ <https://www.cuga-moylan.com/blog/setting-up-eslint-with-airbnb-styleguide-in-sublime-text-3/>)
> For the pull request, if you push your changes to your github repo then it should update this PR automatically.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/muicss/mui-angular/pull/2#issuecomment-689821753>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AM6FHO2SBTAQ4QDDKI3YKYDSE7U2TANCNFSM4RA46NZQ>.
>

jefrog1844 commented:
I applied the changes you requested and got the unit test to work.  I'm
going to see about getting eslint working now.

Jeff

On Wed, Sep 9, 2020 at 2:06 AM Andres Morey <notifications@github.com>
wrote:

> Looks great! I added some suggestions in-line. Two other comments:
>
>    - If possible it would be nice to add some tests as well. I've found
>    that even simple tests can help you to improve your code and make it more
>    maintainable in the long-run.
>    - Are you familiar with Airbnb's JavaScript styleguide
>    <https://github.com/airbnb/javascript>? I use it for my javascript
>    code and have found it to be very useful. It has some tips that would be
>    helpful here too.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/muicss/mui-angular/pull/2#issuecomment-689353881>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM6FHO4DRNBME4GQ7ELZJJTSE4SO7ANCNFSM4RA46NZQ>
> .
>

amorey commented:
Awesome! Looks great! Let me know when you get the linter working.

jefrog1844 commented:
the air bnb suggested additional formatting so I applied those.  Also, found a cleaner way to check class list.  I found that the disabled check appeared to do nothing so I removed it for now.

jefrog1844 commented:
I’m struggling on linter, it just looks too messy as everyone has a different way to include it in their code editor.  I had it working in vs code, but now it is not and it altered angular.json, tsconfig.json, and it conflicted with tslint.json, on and on.

I’m not sure if you want to include lint of dependencies in the library or not.  If you have an easy suggest, I’m open to it.

Also, on GitHub: I made another submit.  From what I understand, I have a forked copy that I can keep updated from your master.  From my fork, I can create a branch like I did for button.  The way I understand it now is that I should wait until you merge the button branch and then fetch from upstream to update my fork.  Then I can create another branch for the next feature.  Correct?

Thanks for the help.

Jeff

> On Sep 10, 2020, at 1:50 AM, Andres Morey <notifications@github.com> wrote:
> 
> 
> Awesome! Looks great! Let me know when you get the linter working.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/muicss/mui-angular/pull/2#issuecomment-690028027>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AM6FHOZ2O2CJVMJMWE5Z7OLSFBZL5ANCNFSM4RA46NZQ>.
>

amorey commented:
Same here. I tried setting up eslint and ran into problems. For now, maybe we should stick with Angular's default linter. Just run `ng lint` and it will make some suggestions.

Your github workflow sounds good (1. create feature branch 2. submit pull request 3. merge upstream changes into master 4. repeat).

jefrog1844 commented:
Sounds good.  I'll test ng lint and I've got a good idea of what you're
looking for now.

What component do you want next?  You'll need to merge the button branch in
first correct?

Thanks

Jeff

On Fri, Sep 11, 2020 at 7:30 AM Andres Morey <notifications@github.com>
wrote:

> Same here. I tried setting up eslint and ran into problems. For now, maybe
> we should stick with Angular's default linter. Just run ng lint and it
> will make some suggestions.
>
> Your github workflow sounds good (1. create feature branch 2. submit pull
> request 3. merge upstream changes into master 4. repeat).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/muicss/mui-angular/pull/2#issuecomment-691065568>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM6FHO224OM3ZWTMB6QC57LSFIKAHANCNFSM4RA46NZQ>
> .
>

amorey commented:
I think you can probably submit another PR independent of the button merge but we might as well merge first. How about knocking off some easy ones next, like container, divider, panel?

> On Sep 11, 2020, at 11:04 PM, jefrog1844 <notifications@github.com> wrote:
> 
> ﻿
> Sounds good. I'll test ng lint and I've got a good idea of what you're
> looking for now.
> 
> What component do you want next? You'll need to merge the button branch in
> first correct?
> 
> Thanks
> 
> Jeff
> 
> On Fri, Sep 11, 2020 at 7:30 AM Andres Morey <notifications@github.com>
> wrote:
> 
> > Same here. I tried setting up eslint and ran into problems. For now, maybe
> > we should stick with Angular's default linter. Just run ng lint and it
> > will make some suggestions.
> >
> > Your github workflow sounds good (1. create feature branch 2. submit pull
> > request 3. merge upstream changes into master 4. repeat).
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/muicss/mui-angular/pull/2#issuecomment-691065568>, or
> > unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/AM6FHO224OM3ZWTMB6QC57LSFIKAHANCNFSM4RA46NZQ>
> > .
> >
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

jefrog1844 commented:
That sounds fine, I will work on them tomorrow night.

BTW,
Nice pick on the ng lint; bonus since it is integrated with angular already
and if anyone else were to contribute we'd all be on the same linter.  I
tested the tslint plugin for vscode which I’ve been using and it lines up
really well with ng lint.  Now I know why the tslint.json files are in the
project folder. 😃

Thanks for your help,

Jeff

On Fri, Sep 11, 2020 at 4:10 PM Andres Morey <notifications@github.com>
wrote:

> I think you can probably submit another PR independent of the button merge
> but we might as well merge first. How about knocking off some easy ones
> next, like container, divider, panel?
>
> > On Sep 11, 2020, at 11:04 PM, jefrog1844 <notifications@github.com>
> wrote:
> >
> > ﻿
> > Sounds good. I'll test ng lint and I've got a good idea of what you're
> > looking for now.
> >
> > What component do you want next? You'll need to merge the button branch
> in
> > first correct?
> >
> > Thanks
> >
> > Jeff
> >
> > On Fri, Sep 11, 2020 at 7:30 AM Andres Morey <notifications@github.com>
> > wrote:
> >
> > > Same here. I tried setting up eslint and ran into problems. For now,
> maybe
> > > we should stick with Angular's default linter. Just run ng lint and it
> > > will make some suggestions.
> > >
> > > Your github workflow sounds good (1. create feature branch 2. submit
> pull
> > > request 3. merge upstream changes into master 4. repeat).
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/muicss/mui-angular/pull/2#issuecomment-691065568>,
> or
> > > unsubscribe
> > > <
> https://github.com/notifications/unsubscribe-auth/AM6FHO224OM3ZWTMB6QC57LSFIKAHANCNFSM4RA46NZQ
> >
> > > .
> > >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub, or unsubscribe.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/muicss/mui-angular/pull/2#issuecomment-691313373>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM6FHO6ADFDLUCUPWQNOZATSFKG3RANCNFSM4RA46NZQ>
> .
>

jefrog1844 commented:
I modified tslint to stop modifying/removing the optional parameter on properties when we set a default value.  Now we can set default values that are optional as well.  Also removed the requirement to have the "lib" prefix as we're not using that.  Linting is working great now.

jefrog1844 commented:
Squashed the multiple commits for the 'button' branch
